src/api/app/models/branch_package.rb

```diff
@@ -127,10 +127,8 @@ def create_branch_packages(tprj)
       tpkg.store
 
       if p[:local_link]
-        # rubocop:disable Metrics/LineLength
         # copy project local linked packages
-        Backend::Connection.post "/source/#{tpkg.project.name}/#{tpkg.name}?cmd=copy&oproject=#{CGI.escape(p[:link_target_project].name)}&opackage=#{CGI.escape(p[:package].name)}&user=#{CGI.escape(User.current.login)}"
-        # rubocop:enable Metrics/LineLength
+        Backend::Api.copy_package(tpkg.project.name, tpkg.name, p[:link_target_project].name, p[:package].name, User.current.login)
```

Comment:
This makes the code less readable IMHO, since you don't know which parameters are used for what. Is the project name for origin or release project for example?

Can you refactor it in that way, that you hand over a hash? That makes also the order of the parameters unimportant.

Reply:
![imagen](https://user-images.githubusercontent.com/11314634/29816130-402c8558-8cb3-11e7-9a72-a2cd5a979573.png)

> This makes the code less readable IMHO, since you don't know which parameters are used for what.

Beside that the params are the same as before... for knowing what the params are for without taking a look at the method implementation you only need to use good names for the variables you pass, for example:

```ruby
 Backend::Api.copy_package(target_project, target_package, source_project, source_package, User.current.login)
```

> Can you refactor it in that way, that you hand over a hash?

I can, but I don't really think this is needed. I mean, using a hash is because normally you want to pass optional params (and values/variables passed are not self explanatory) or a structured data that can vary from one call to another and I don't think this is the case. For example, if you have an optional param that is meant to be a boolean:

```ruby
def my_method(project, options = {})
#... 
project.destroy if options[:destroy]
#...
end
# that way you know what is the second/optional param for
my_method(project, destroy: true)
# in comparison to this, that you don't know what is that for
my_method(project, true)
```
In our case, it have 5 params and all must be there.

And also if I use a hash then I have to code checks to be sure that everything is passed and raise exceptions if not (or fail with kind of a code). With "normal¨ params the check is coming from the interpreter.

> That makes also the order of the parameters unimportant.

Why is that better? I mean, do you expect people to randomize the order of the params in  method calls? Most of the programming languages have order for the params... and nobody changed that.

Reply:
I agree with @mdeniz here. I find with his option much more readable. Which parameters are used for what is set in the method definition. I don't see a point in using a hash.

Reply:
okay, you convinced me here if this is limited to the copy operation... all the other paramters (>10 IIRC) would still be handed over via a hash, right?

(The old code should have user tpkg.source_path btw)

Reply:
NO in general!!!! I'm just translating what we call in the frontend. 

I mean, if we use 5 params that's it, if we have 3 that's it... I'm not talking about what params the backend could eat in a singular endpoint. I'm talking about what we pass from the frontend. So, in general we should create Backend::Api methods that have the exact number of params needed to do what is meant to be done in the backend.

For the source controller of the frontend api we will have that problem, that just passes whatever is coming in the url to the backend... we will have to deal with that, and maybe in that case we will use a hash because we don't know in advance what params to expect.

Reply:
> The old code should have user tpkg.source_path btw

We are also removing the need to calculate urls in the frontend models, so that source_path will be calculated in the corresponding Backend::Api method